transport/stream.go

```diff
@@ -53,7 +53,7 @@ func (dc *duplexConnAdaptor) Write(b []byte) (int, error) {
 }
 func (dc *duplexConnAdaptor) ReadFrom(r io.Reader) (int64, error) {
 	// Make sure we prefer ReadFrom. Otherwise io.Copy will try WriteTo first.
-	if rf, ok := r.(io.ReaderFrom); ok {
+	if rf, ok := dc.w.(io.ReaderFrom); ok {
```

Comment:
Can we add some tests for `duplexConnAdaptor`, to ensure that we won't introduce any regressions in the future.

Reply:
done